### PR TITLE
Fix textbox direction when it contains both RTL and LTR languages

### PR DESCRIPTION
**Pull request by mokazemi**
**Beofre:**

1. In a text box, text direction was set to Left-to-Right, regardless of the text language.

**After:**
1. Text box direction is set to auto, so that it matches the direction of text language. 
2. Text dir is set to auto for each line of the text. To achieve this, each line of text is now put in separated `<span>` tags with `dir="auto"`.

Screenshot_20240318_152725

Note: Without putting each line of text into separated `<span>` tags, whole text direction was in direction of initial line text. So each line of text is now wrapped into separate tags.

### Change Type

- [ ] `sdk` — Changes the tldraw SDK
- [x] `dotcom` — Changes the tldraw.com web app
- [ ] `docs` — Changes to the documentation, examples, or templates.
- [ ] `vs code` — Changes to the vscode plugin
- [ ] `internal` — Does not affect user-facing stuff

- [x] `bugfix` — Bug fix
- [ ] `feature` — New feature
- [x] `improvement` — Improving existing features
- [ ] `chore` — Updating dependencies, other boring stuff
- [ ] `galaxy brain` — Architectural changes
- [ ] `tests` — Changes to any test code
- [ ] `tools` — Changes to infrastructure, CI, internal scripts, debugging tools, etc.
- [ ] `dunno` — I don't know

### Test Plan

Enter a text right-to-left and left-to-right into textboxes and note the change of direction by clicking outside of the box, before and after this PR.

### Release Notes

Fix textbox direction when it contains both RTL and LTR languages

### CODE REVIEW

**Comment by mimecuvalo** [comment]
Thanks for the PR! We're actively thinking about our textfields at the moment so this is good timing on your part.

Lemme keep this PR in a holding pattern for now because I want the dust to settle on some of the other work first, but this is great initial direction and this will help with our RTL/LTR mechanics. 👍

**Comment by steveruizok** [comment]
I'd be ok to land this but without the spans. If there's mixed RTL, I'd defer to the browser to decide which direction to display the text.

**Comment by mokazemi** [comment]
> I'd be ok to land this but without the spans. If there's mixed RTL, I'd defer to the browser to decide which direction to display the text.

The problem is browser handles `textarea` and `div` differently when containing multi-line texts with combination of RTL and LTR languages. In `textarea`, every line's direction is considered separately, but in `div` or `span`, whole text's direction is set according to the first line. 
So you'll see different behavior while editing the text and after finishing editing, which is not acceptable. This is the video to show the issue:

[Screencast_20240417_224019.webm](https://github.com/tldraw/tldraw/assets/18334056/7d052643-9a44-4b52-9a2c-1ed2e3d67218)

Putting each line in different tags will solve the issue.

File: packages/tldraw/src/lib/shapes/shared/TextLabel.tsx [hunk 1]
```diff
@@ -54,8 +54,16 @@ export const TextLabel = React.memo(function TextLabel<
...
 	} = useEditableText(id, type, text)
 
-	const finalText = TextHelpers.normalizeTextForDom(text)
-	const hasText = finalText.length > 0
+	const finalTextArray = TextHelpers.normalizeTextForDom(text).split('\n')
+	const hasText = finalTextArray.length > 0
+
+	// create final text by putting each line into separate tags with dir="auto" to fix bidi-text direction
+	const finalText = finalTextArray.map((lineOfText, index) => (
+		<span dir="auto" key={index}>
```
[the remark this answers is not in the document]
Comment (mokazemi) [reply packages/tldraw/src/lib/shapes/shared/TextLabel.tsx:62]: Thanks for suggesting. I'll check it tomorrow, but I guess it works too. As you said, we can get rid of `<br />` if we use `<div>`.

**Comment by steveruizok** [comment]
I'm not a RTL language user, how common is it for the different directional languages to be combined?

**Comment by mimecuvalo** [comment]
> I'm not a RTL language user, how common is it for the different directional languages to be combined?

Yeah, I've spearheaded internationalization before and yeah, it's common. We need to support it because that would make for more Perfect Text in canvas ;)

**Review by mimecuvalo — APPROVED** [review]
LGTM

**Pull request merged**